crates/aptos-faucet/core/src/funder/mint.rs

```diff
@@ -290,9 +363,46 @@ impl FunderTrait for MintFunder {
         &self,
         amount: Option<u64>,
         receiver_address: AccountAddress,
+        asset: Option<String>,
         check_only: bool,
         did_bypass_checkers: bool,
     ) -> Result<Vec<SignedTransaction>, AptosTapError> {
+        // Resolve asset (use configured default if not specified)
+        let asset_name = asset.as_deref().unwrap_or(&self.default_asset);
+
+        // Get asset config
+        let asset_config = self.assets.get(asset_name).ok_or_else(|| {
+            AptosTapError::new(
+                format!("Asset '{}' is not configured", asset_name),
+                AptosTapErrorCode::InvalidRequest,
+            )
+        })?;
+
+        // Only set faucet_account if it's NOT the default asset
+        // (default asset is already set and delegated at startup)
+        if asset_name != self.default_asset {
+            let key = asset_config.get_key().map_err(|e| {
+                AptosTapError::new_with_error_code(e, AptosTapErrorCode::InvalidRequest)
+            })?;
+            let account_address = asset_config.mint_account_address.unwrap_or_else(|| {
+                AuthenticationKey::ed25519(&Ed25519PublicKey::from(&key)).account_address()
+            });
+
+            let mint_account = LocalAccount::new(account_address, key, 0);
+            {
+                let mut faucet_account = self.faucet_account.write().await;
+                *faucet_account = mint_account;
```

Comment:
This pattern is really weird and maybe even unsafe (I need to think about it harder). We should just create a LocalAccount per asset, and only once at startup right?